src/backend/distributed/utils/maintenanced.c

```diff
@@ -248,7 +253,8 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 	{
 		int rc;
 		int latchFlags = WL_LATCH_SET | WL_TIMEOUT | WL_POSTMASTER_DEATH;
-		int timeout = 10000; /* wake up at least every so often */
+		double timeout = 10000.0; /* use this if the deadlock detection is disabled */
```

Comment:
Doing the calculations with `double` seems better because we might divide the value? 

I added explicit cast on `WaitLatch` and a single line comment. Does that work?